packages/web/components/one-click-trading/one-click-trading-settings.tsx

```diff
@@ -147,7 +176,9 @@ export const OneClickTradingSettings = ({
   useEffect(() => {
     if (!transaction1CTParams || initialTransaction1CTParams) return;
     setInitialTransaction1CTParams(transaction1CTParams);
-  }, [initialTransaction1CTParams, transaction1CTParams]);
+    setChanges(externalChanges);
+    setInitialChanges(externalChanges);
```

Comment:
To avoid maintaining two sources of truth, I recommend using controlled inputs to override the local changes state when necessary. We have a handy hook called useControlledState that simplifies this process. For documentation, refer to: [Chakra UI - useControllable Hook](https://v2.chakra-ui.com/docs/hooks/use-controllable).

Reply:
Wow, this is a nice pattern! I will be using this in the future, for sure!

But it does not work in this particular case. These are not two sources of truth but a draft and a draft of a draft. 
We create a draft, the `transaction1CTParams`, with its changes. This stores the value set by the user until the confirm button is pressed. 

When the user opens the edit modal, another draft is created, which can be saved by pressing back and discarded by pressing `X`.  This lets the user edit the parameters multiple times before sending the actual tx. 

The implementation is optimistic — we edit the main draft during the process by default but revert the changes and the main draft to the snapshot they were in when the modal was opened if they user decides to discard the changes.

![CleanShot 2024-11-19 at 19 33 32](https://github.com/user-attachments/assets/bab5787b-d9e5-405a-8a64-386427355148)


If I misunderstood what you meant and the `useControlledState` makes sense, please let me know and I will try to make it work